### PR TITLE
callFunction0 pointer param portability to 32-bit system

### PR DESCRIPTION
**Pull request by tmandys**

### CODE REVIEW

**Comment by davidnich** [comment]
@tmandys please assign to me when ready to merge

File: docs/mainpage.dox.tmpl [hunk 1]
```diff
@@ -317,4 +317,5 @@ System::out.println("testing...");
 
     @subsection jni_1_0 jni Module Version 1.0
     - initial public release
+    - fixed callFunction0 issue when running at 32-bit system
```
Comment (davidnich) [inline docs/mainpage.dox.tmpl]: please remove this release note - this is unreleased code - release notes are meant to document changes between releases

**Review by davidnich — CHANGES_REQUESTED** [review]

**Comment by omusil24** [comment]
@tmandys Can you explain why passing a simple pointer doesn't work?

**Comment by tmandys** [comment]
Because parameter is defined as jlong (64bit) but (Class*) pointer is 32-bit at 32bit OS so the calling function stack prepared by java does not correrspond to stack expecting by C++. I.e. use jlong parameter and retype to pointer in next step. It was already used in another function so it is not my invention.

**Comment by omusil24** [comment]
Ok. Thank you.

**Pull request merged**